apollo-router/src/router_factory.rs

```diff
@@ -258,6 +258,17 @@ impl YamlRouterFactory {
                         .experimental_reuse_query_plans,
                 )
                 .await;
+        } else {
+            supergraph_creator
+                .prewarm_query_planner(
+                    &query_analysis_layer,
+                    &persisted_query_layer,
+                    configuration
+                        .supergraph
+                        .query_planning
+                        .experimental_reuse_query_plans,
+                )
+                .await;
```

Comment:
maybe something like this would be simpler:

```rust
let previous_cache = previous_router.map(|router| previous_router.previous_cache());
supergraph_creator
                .warm_up_query_planner(
                    &query_analysis_layer,
                    &persisted_query_layer,
                    previous_cache,
                    configuration.supergraph.query_planning.warmed_up_queries,
                    configuration
                        .supergraph
                        .query_planning
                        .experimental_reuse_query_plans,
                )
                .await;
```

While having `warm_up_query_planner` take an `Option<InMemoryCache>` as argument

Reply:
Yeah, I had considered that so that sounds good; felt like prewarm was just a little clearer for now, but can use the same path for both cases.